warp10/src/main/java/io/warp10/script/functions/MOTIONSPLIT.java

```diff
@@ -0,0 +1,205 @@
+   * 
+   * @param gts GTS to split
+   * @param label Name of label which will contain the start of the split
+   * @param timeThreshold If the delay between two ticks goes beyond this value, force a split
+   * @param distanceThreshold If we traveled more than distanceThreshold between two ticks, force a split
+   * @param proximityZoneRadius Radius of the proximity zone in meters
+   * @param proximityZoneTime If we spent more than that much time in the proximity zone without moving and traveled slower than proximityZoneSpeed while in the proximityZone force a split
+   * @param proximityZoneSpeed Minimum speed in the proxmityZone to prevent a split, in m/s
+   * 
+   * @return
+   * @throws WarpScriptException
+   */
+  private static List<GeoTimeSerie> motionSplit(GeoTimeSerie gts, String label, long timeThreshold, double distanceThreshold, double proximityZoneRadius, double proximityZoneSpeed, long proximityZoneTime) throws WarpScriptException {
+    //
+    // Sort GTS according to timestamps
+    //
+    
+    GTSHelper.sort(gts, false);
+
+    GeoTimeSerie split = null;
+    
+    List<GeoTimeSerie> splits = new ArrayList<GeoTimeSerie>();
+    
+    int idx = 0;
+    
+    int n = GTSHelper.nvalues(gts);
+    
+    boolean mustSplit = true;
+
+    long refLocation = GeoTimeSerie.NO_LOCATION;
+    long refTimestamp = Long.MIN_VALUE;
+    
+    long previousTimestamp = Long.MIN_VALUE;
+    long previousLocation = GeoTimeSerie.NO_LOCATION;
+    double proximityZoneTraveledDistance = 0.0D;
+    
+    while (idx < n) {
+      long timestamp = GTSHelper.tickAtIndex(gts, idx);
+      long location = GTSHelper.locationAtIndex(gts, idx);
+      long elevation = GTSHelper.elevationAtIndex(gts, idx);
+      Object value = GTSHelper.valueAtIndex(gts, idx);
+    
+      //
+      // If the previous tick was more than 'timeThreashold' ago, split now
+      // If the distance to the previous location is above distanceThreshold, split
+      //
+      
+      if (timestamp - previousTimestamp > timeThreshold
+          || (GeoTimeSerie.NO_LOCATION != previousLocation && GeoTimeSerie.NO_LOCATION != location && GeoXPLib.orthodromicDistance(location, previousLocation) > distanceThreshold)) {
+        mustSplit = true;
+      }
+      
+      //
+      // If the current point is farther away from the reference point than 'proximityZoneRadius'
+      // change the reference point and reset the traveled distance.
+      // If the distance traveled in the proximity zone was traveled at less than proximityZoneSpeed and the time spent in the zone is above proximityZoneTime, force a split
+      //
+      
+      if (GeoTimeSerie.NO_LOCATION != refLocation && GeoTimeSerie.NO_LOCATION != location && GeoXPLib.orthodromicDistance(refLocation, location) > proximityZoneRadius) {
+        if (previousTimestamp - refTimestamp > proximityZoneTime && proximityZoneTraveledDistance / ((previousTimestamp - refTimestamp) / Constants.TIME_UNITS_PER_S) < proximityZoneSpeed) {
```

Comment:
`(previousTimestamp - refTimestamp) / Constants.TIME_UNITS_PER_S` is still an integer division which could, if timestamps are very close, lead to a division by 0.